Formula/cloudprober.rb

```diff
@@ -0,0 +1,37 @@
+class Cloudprober < Formula
+  desc "Active monitoring software to detect failures before your customers do"
+  homepage "https://cloudprober.org"
+  url "https://github.com/cloudprober/cloudprober/archive/refs/tags/v0.12.0.tar.gz"
+  sha256 "e2289eaad263e35e20ed3afaccf4c854df179de4e36aa705fcf106d93a352b0b"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "make", "cloudprober", "VERSION=v#{version}"
+    bin.install "cloudprober"
+  end
+
+  test do
+    require "pty"
+    require "timeout"
+
+    reader, writer, pid, = PTY.spawn(bin/"cloudprober --logtostderr 2>&1")
+
+    listening = Timeout.timeout(10) do
+      found = false
+      reader.each do |line|
+        if /Initialized status surfacer/.match?(line)
+          found = true
+          break
+        end
+      end
+      found
+    end
+
+    Process.kill("TERM", pid)
+    writer.close
+    reader.close
+    listening
+  end
```

Comment:
Can you see if either of these would work?

```suggestion
  test do
    io = IO.popen("#{bin}/cloudprober", "--logtostderr", err: [:child, :out])
    io.any? do |line|
      /Initialized status surfacer/.match?(line)
    end
  end
```

```suggestion
  test do
    require "open3"

    Open3.popen3("#{bin}/cloudprober --logtostderr 2>&1") do |_, stdout|
      stdout.any? do |line|
        /Initialized status surfacer/.match?(line)
      end
    end
  end
```

As-is the test is a bit unwieldy. Things that may help frame the above:
- no `require` is preferred if possible
- I'm unsure if `Initialized status surfacer` is the first line output/last line output/etc. If it's the first line output: the above can probably be simplified
- our `test do` already should handle timeouts, closing and killing processes. Please check and see if I'm wrong here and only include them if required

Thanks again!

Reply:
(Looks like my earlier message got lost.)

Thanks a lot for the suggestions. I verified that first suggestion works.